packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts

```diff
@@ -0,0 +1,225 @@
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EksCreateClusterProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EksCreateCluster.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: ['*'],
+        actions: ['eks:CreateCluster'],
+      }),
+      new iam.PolicyStatement({
+        actions: ['iam:PassRole'],
+        resources: [props.role],
+        conditions: {
+          StringEquals: { 'iam:PassedToService': 'eks.amazonaws.com' },
+        },
+      }),
+    ];
+  }
+
+  /**
+   * Provides the EKS create cluster service integration task configuration
+   */
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    if (this.props.loggingConfiguration?.clusterLoggingList) {
+      return {
+        Resource: integrationResourceArn('eks', 'createCluster', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          Name: this.props.name,
+          Version: this.props.kubernetesVersion,
+          RoleArn: this.props.role,
+          ResourcesVpcConfig: {
+            SubnetIds: this.props.resourcesVpcConfig.subnetIds,
+            SecurityGroupIds: this.props.resourcesVpcConfig.securityGroupIds,
+            EndpointPublicAccess: this.props.resourcesVpcConfig.endpointPublicAccess,
+            EndpointPrivateAccess: this.props.resourcesVpcConfig.endpointPrivateAccess,
+            PublicAccessCidrs: this.props.resourcesVpcConfig.publicAccessCidrs,
+          },
+          Logging: {
+            ClusterLogging: this.props.loggingConfiguration?.clusterLoggingList,
+          },
+          ClientRequestToken: this.props.clientRequestToken,
+          Tags: this.props.tags,
+          EncryptionConfig: this.props.encryptionConfig,
+        }),
+      };
+    } else {
+      return {
+        Resource: integrationResourceArn('eks', 'createCluster', this.integrationPattern),
+        Parameters: sfn.FieldUtils.renderObject({
+          Name: this.props.name,
+          Version: this.props.kubernetesVersion,
+          RoleArn: this.props.role,
+          ResourcesVpcConfig: {
+            SubnetIds: this.props.resourcesVpcConfig.subnetIds,
+            SecurityGroupIds: this.props.resourcesVpcConfig.securityGroupIds,
+            EndpointPublicAccess: this.props.resourcesVpcConfig.endpointPublicAccess,
+            EndpointPrivateAccess: this.props.resourcesVpcConfig.endpointPrivateAccess,
+            PublicAccessCidrs: this.props.resourcesVpcConfig.publicAccessCidrs,
+          },
+          ClientRequestToken: this.props.clientRequestToken,
+          Tags: this.props.tags,
+          EncryptionConfig: this.props.encryptionConfig,
+        }),
+      };
+    }
+  }
```

Comment:
there's a fair amount of duplication here - let's simplify it down to one block.